src/Stratis.Bitcoin/Utilities/NodeRunningLock.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.IO;
+using Stratis.Bitcoin.Configuration;
+
+namespace Stratis.Bitcoin.Utilities
+{
+    /// <summary>
+    /// Class that prevents another instance of the node to run in the same data folder
+    /// and allows external applications to see if the node is running.
+    /// </summary>
+    public class NodeRunningLock
+    {
+        private readonly string lockFileName;
+
+        private FileStream fileStream;
+
+        public NodeRunningLock(DataFolder dataFolder)
+        {
+            this.lockFileName = Path.Combine(dataFolder.RootPath, "lockfile");
+        }
+
+        public bool TryLockNodeFolder()
+        {
+            try
+            {
+                this.fileStream = new FileStream(this.lockFileName, FileMode.Create, FileAccess.ReadWrite, FileShare.None);
+                return true;
+            }
+            catch (IOException)
+            {
+                return false;
+            }
+        }
+
+        public void UnlockNodeFolder()
+        {
+            this.fileStream.Close();
+            File.Delete(this.lockFileName);
```

Comment:
You probably need to loop here until the file is confirmed to be deleted

Reply:
Why do we need to confirm it's deleted? 
Once `File.Delete` is called we know that file will be deleted. But we have no reason to wait till file system actually does it

Reply:
To my recollection this does not always work, is can be problematic.

You want to loop on if file exists and put a time delay in between.

Reply:
I'm pretty sure that File.Delete works as long as there is no readonly attribute and file is not in some system folder. It can prematurely exit the method before the file is deleted but it will be deleted. 

At least I didnt find any info that would state otherwise.